executor/utils.go

```diff
@@ -53,3 +58,201 @@ func deleteFromSet(set []string, value string) []string {
 	}
 	return set
 }
+
+// SQLDigestTextRetriever is used to find the normalized SQL statement text by SQL digests in statements_summary table.
+// It's exported for text purposes.
+type SQLDigestTextRetriever struct {
+	// SQLDigestsMap is the place to put the digests that's requested for getting SQL text and also the place to put
+	// the query result.
+	SQLDigestsMap map[string]string
+
+	// Replace querying for test purposes.
+	mockLocalData  map[string]string
+	mockGlobalData map[string]string
+	// There are two ways for querying information: 1) query specified digests by WHERE IN query, or 2) query all
+	// information to avoid the too long WHERE IN clause. If there are more than `fetchAllLimit` digests needs to be
+	// queried, the second way will be chosen; otherwise, the first way will be chosen.
+	fetchAllLimit int
+}
+
+// NewSQLDigestTextRetriever creates a new SQLDigestTextRetriever.
+func NewSQLDigestTextRetriever() *SQLDigestTextRetriever {
+	return &SQLDigestTextRetriever{
+		SQLDigestsMap: make(map[string]string),
+		fetchAllLimit: 512,
+	}
+}
+
+func (r *SQLDigestTextRetriever) runMockQuery(data map[string]string, inValues []interface{}) (map[string]string, error) {
+	if len(inValues) == 0 {
+		return data, nil
+	}
+	res := make(map[string]string, len(inValues))
+	for _, digest := range inValues {
+		if text, ok := data[digest.(string)]; ok {
+			res[digest.(string)] = text
+		}
+	}
+	return res, nil
+}
+
+// runQuery runs query to the system tables to fetch normalized SQL texts corresponding to given SQL digests, if
+// `inValues` is given, or all digest -> SQL mappings otherwise. If `queryGlobal` is false, it queries
+// information_schema.statements_summary and information_schema.statements_summary_history; otherwise, it queries the
+// cluster version of these two tables.
+func (r *SQLDigestTextRetriever) runQuery(ctx context.Context, sctx sessionctx.Context, queryGlobal bool, inValues []interface{}) (map[string]string, error) {
```

Comment:
How about `runFetchDigestQuery`, `runQuery` seems to be a general query executor.